internal/controller/issues.go

```diff
@@ -0,0 +1,171 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/v56/github"
+	issues "github.com/matanamar10/github-issue-operator-hhome-assignment/api/v1alpha1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"time"
+)
+
+// searchForIssue checks if GithubIssue CRD has an issue in the repo
+func searchForIssue(issue *issues.GithubIssue, gitHubIssues []*github.Issue) *github.Issue {
+	for _, ghIssue := range gitHubIssues {
+		if ghIssue != nil && strings.EqualFold(*ghIssue.Title, issue.Spec.Title) {
+			return ghIssue
+		}
+	}
+	return nil
+}
```

Comment:
Your code is gonna fail if you use a different `git` implementation like `gitlab` since it still relies on concrete GitHub types. Use interfaces, move the code to packages and files where it makes sense in terms of different layers.